### DISCUSSION TITLE
About Font.

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [marcomusy/vedo#815] [Q&A] [answered]
Hi Marco,
I want to use 'LogoType' Font but I got this:
Could you tell me why? Thanks!
> [vedo.shapes.py:3874] WARNING: font name LogoType not found.
> Traceback (most recent call last):
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\numpy\lib\npyio.py", line 441, in load
>     return pickle.load(fid, **pickle_kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> EOFError: Ran out of input
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\vedo\shapes.py", line 3872, in _load_font
>     font_meshes = np.load(fontfile, allow_pickle=True)["font"][0]
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\numpy\lib\npyio.py", line 443, in load
>     raise pickle.UnpicklingError(
> _pickle.UnpicklingError: Failed to interpret file 'C:\\Users\\xuzig\\AppData\\Local\\Temp\\LogoType.npz' as a pickle
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "d:\BaiduNetdiskWorkspace\XVedo\test\test_axes_def.py", line 13, in <module>
>     axes = Axes(
>            ^^^^^
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\vedo\addons.py", line 3497, in Axes
>     xt = shapes.Text3D(
>          ^^^^^^^^^^^^^^
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\vedo\shapes.py", line 3983, in __init__
>     tpoly = self._get_text3d_poly(
>             ^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\vedo\shapes.py", line 4152, in _get_text3d_poly
>     poly = _get_font_letter(font, t)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\vedo\shapes.py", line 3882, in _get_font_letter
>     font_meshes = _load_font(font)
>                   ^^^^^^^^^^^^^^^^
>   File "C:\Users\xuzig\AppData\Local\Programs\Python\Python311\Lib\site-packages\vedo\shapes.py", line 3875, in _load_font
>     raise RuntimeError
> RuntimeError

**Comment by marcomusy** [comment, marked as answer]
Uhm are you sure you have a valid internet connection? `LogoType` (like many other fonts) is downloaded on the fly and cached in a temporary directory.

**Reply by ZiguoAtGitHub** [reply]
Sigh......The abroad internet connection has some filter measures in our country.
I will re-try in next days. Maybe the internet connection could be ok or I should give up those fonts.
Anyway, thank you again!

**Reply by marcomusy** [reply]
Maybe you can copy them locally and give the local path?
https://vedo.embl.es/fonts/LogoType.ttf
https://vedo.embl.es/fonts/LogoType.npz

**Reply by ZiguoAtGitHub** [reply]
Thank you, Marco.
You are so generous!
